polymetis/polymetis/src/polymetis_server.cpp

```diff
@@ -292,17 +295,28 @@ Status PolymetisControllerServerImpl::UpdateController(
 
   // Update controller & set intervals
   if (custom_controller_context_.status == RUNNING) {
-    custom_controller_context_.controller_mtx.lock();
-    interval->set_start(robot_state_buffer_.size());
-    custom_controller_context_.custom_controller->param_dict_update_module();
-    custom_controller_context_.controller_mtx.unlock();
+    try {
+      custom_controller_context_.controller_mtx.lock();
+      interval->set_start(robot_state_buffer_.size());
+      custom_controller_context_.custom_controller->param_dict_update_module();
+      custom_controller_context_.controller_mtx.unlock();
+
+    } catch (const std::exception &e) {
+      custom_controller_context_.controller_mtx.unlock();
+
+      std::string error_msg =
+          "Failed to update controller: " + std::string(e.what());
+      spdlog::error(error_msg);
+      return Status(StatusCode::CANCELLED, error_msg);
+    }
+
   } else {
-    spdlog::warn(
-        "Tried to perform a controller update with no controller running.");
-    interval->set_start(-1);
-  }
+    std::string error_msg =
+        "Tried to perform a controller update with no controller running.";
+    spdlog::warn(error_msg);
```

Comment:
Relatedly & in response to Giri's request, I think this and line 340 should actually throw on the clientside & interrupt